pkg/operator/controller/ingress/rsyslog_configmap.go

```diff
@@ -43,25 +43,30 @@ func (r *reconciler) ensureRsyslogConfigMap(ic *operatorv1.IngressController, de
 		return false, nil, nil
 	case !wantCM && haveCM:
 		if deleted, err := r.deleteRsyslogConfigMap(current); err != nil {
-			return true, current, fmt.Errorf("failed to delete configmap: %v", err)
+			if !errors.IsNotFound(err) {
```

Comment:
`deleteRsyslogConfigMap` returns `false, nil` if `errors.IsNotFound(err)`.  It might be best to get rid of `deleteRsyslogConfigMap` and inline its logic here for consistency.  (The reason that the deletion logic is inconsistent is that we—mostly I—have waffled back and forth on using a separate method for `deleteFoo` versus inlining the logic.)

Reply:
Sure. I agree that inline deletion makes more sense here.